crates/biome_js_analyze/src/analyzers/nursery/use_regex_literals.rs

```diff
@@ -0,0 +1,251 @@
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make::js_regex_literal_expression;
+use biome_js_semantic::SemanticModel;
+use biome_js_syntax::{
+    global_identifier, AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression,
+    AnyJsTemplateElement, JsCallArguments, JsCallExpression, JsComputedMemberExpression,
+    JsNewExpression, JsSyntaxKind, JsSyntaxToken,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, AstNodeList, AstSeparatedList, BatchMutationExt, SyntaxError,
+};
+
+use crate::{semantic_services::Semantic, JsRuleAction};
+
+declare_rule! {
+    /// Enforce the use of the regular expression literals instead of the RegExp constructor if possible.
+    ///
+    /// It is considered a best practice to avoid the string literal notation on top of the regular expression notation,
+    /// and use regular expression literals instead of the constructor function.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/prefer-regex-literals/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// new RegExp("abc", "u");
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// /abc/u;
+    ///
+    /// new RegExp("abc", flags);
+    /// ```
+    ///
+    pub(crate) UseRegexLiterals {
+        version: "1.3.0",
+        name: "useRegexLiterals",
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+declare_node_union! {
+    pub(crate) JsNewOrCallExpression = JsNewExpression | JsCallExpression
+}
+
+pub struct UseRegexLiteralsState {
+    pattern: String,
+    flags: Option<String>,
+}
+
+impl Rule for UseRegexLiterals {
+    type Query = Semantic<JsNewOrCallExpression>;
+    type State = UseRegexLiteralsState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+
+        let (callee, arguments) = parse_node(node)?;
+        if !is_regexp_object(callee, model) {
+            return None;
+        }
+
+        let args = arguments.args();
+        if args.len() > 2 {
+            return None;
+        }
+        let mut args = args.iter();
+
+        let pattern = args.next()?;
+        let pattern = create_pattern(pattern, model)?;
+
+        let flags = match args.next() {
+            Some(flags) => {
+                let flags = create_flags(flags)?;
+                Some(flags)
+            }
+            None => None,
+        };
+        Some(UseRegexLiteralsState { pattern, flags })
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            ctx.query().range(),
+            markup! {
+                "Use a regular expression literal instead of the "<Emphasis>"RegExp"</Emphasis>" constructor."
+            },
+        ))
```

Comment:
We could [improve the diagnostic](https://docs.rs/biome_analyze/latest/biome_analyze/#what-a-rule-should-say-to-the-user) by adding a note that explains why a regex literal is preferred.

```suggestion
        ).note(markup! {
            "Regular expression literals avoid some escaping required in a string literal, and are easier to analyze statically."
        }))
```